### PR TITLE
Change library names to namespaced targets

### PR DESCRIPTION
**Pull request by ffoulkes**
- Replace names of external libraries with namespaced references to external targets.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -99,12 +101,12 @@ endfunction()
 # Google libraries
 function(add_stratum_google_libs _TGT)
     target_link_libraries(${_TGT} PUBLIC
-        glog
-        gflags
-        gpr
-        grpc
-        protobuf
-        grpc++
+        gflags::gflags_shared
```
Comment (5abeel) [inline CMakeLists.txt:104]: This looks like it is being changed from `gflags` to `gflags::gflags_shared` -- not just a namespace change, but also a different target link library. Is this correct/intentional/non-breaking?

Comment (ffoulkes) [reply CMakeLists.txt:104]: `gflags` is the stem of the library name (`libgflags.so*` or `libgflags.a`). cmake specifies `-lgflags` on the compiler/linker command line, and that's it.

`gflags::gflags_shared` is a cmake target defined by the gflags config module. We don't know the actual name of the library, but cmake does. It's one of the properties of the target, as are its actual location, the location of its header files, and its direct dependencies.

We've moved from specifying a string of text to importing a cmake target defined, not by us, but by a module provided by the gflags implementation. The reference is loosely coupled, feature-rich, and abstract. If there compile-time or link-time changes with future gflags implementations, they will be largely transparent to us.

**Review by nupuruttarwar — APPROVED** [review]
LGTM

**Review by 5abeel — APPROVED** [review]
LGTM

**Pull request merged**